routing/supernode/client.go

```diff
@@ -79,13 +81,23 @@ func (c *Client) GetValue(ctx context.Context, k u.Key) ([]byte, error) {
 }
 
 func (c *Client) Provide(ctx context.Context, k u.Key) error {
+	defer log.EventBegin(ctx, "provide", &k).Done()
 	msg := pb.NewMessage(pb.Message_ADD_PROVIDER, string(k), 0)
-	// TODO wrap this to hide the dialer and the local/remote peers
-	msg.ProviderPeers = pb.PeerInfosToPBPeers(c.dialer, []peer.PeerInfo{peer.PeerInfo{ID: c.local}}) // FIXME how is connectedness defined for the local node
-	return c.proxy.SendMessage(ctx, msg)                                                             // TODO wrap to hide remote
+	// FIXME how is connectedness defined for the local node
```

Comment:
> how is connectedness defined for the local node

connectedness is still whether the node is directly connected to another, at the time of request.